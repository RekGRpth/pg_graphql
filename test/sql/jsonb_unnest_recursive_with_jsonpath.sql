with doc(x) as (
    select '{"id": {"eq": "DescNullsFirst"}, "x": [1, {"f": ["a"]},3]}'::jsonb
)
select
    jkr.jpath,
    jkr.obj,
    jsonb_path_query(doc.x, jpath::jsonpath),
    -- Should always be true
    jkr.obj = jsonb_path_query(doc.x, jpath::jsonpath) is_equal
from
    doc,
    graphql.jsonb_unnest_recursive_with_jsonpath(doc.x) jkr(jpath, obj)
