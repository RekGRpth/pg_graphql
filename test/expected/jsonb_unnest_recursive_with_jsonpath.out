with doc(x) as (
    select '{"id": {"eq": "DescNullsFirst"}, "x": [1, {"f": ["a"]},3]}'::jsonb
)
select
    jkr.jpath,
    jkr.obj,
    jsonb_path_query(doc.x, jpath::jsonpath),
    -- Should always be true
    jkr.obj = jsonb_path_query(doc.x, jpath::jsonpath) is_equal
from
    doc,
    graphql.jsonb_unnest_recursive_with_jsonpath(doc.x) jkr(jpath, obj)
      jpath      |                             obj                             |                      jsonb_path_query                       | is_equal 
-----------------+-------------------------------------------------------------+-------------------------------------------------------------+----------
 $               | {"x": [1, {"f": ["a"]}, 3], "id": {"eq": "DescNullsFirst"}} | {"x": [1, {"f": ["a"]}, 3], "id": {"eq": "DescNullsFirst"}} | t
 $."id"          | {"eq": "DescNullsFirst"}                                    | {"eq": "DescNullsFirst"}                                    | t
 $."id"."eq"     | "DescNullsFirst"                                            | "DescNullsFirst"                                            | t
 $."x"           | [1, {"f": ["a"]}, 3]                                        | [1, {"f": ["a"]}, 3]                                        | t
 $."x"[0]        | 1                                                           | 1                                                           | t
 $."x"[1]        | {"f": ["a"]}                                                | {"f": ["a"]}                                                | t
 $."x"[1]."f"    | ["a"]                                                       | ["a"]                                                       | t
 $."x"[1]."f"[0] | "a"                                                         | "a"                                                         | t
 $."x"[2]        | 3                                                           | 3                                                           | t
(9 rows)

